airflow/models/dag.py

```diff
@@ -2884,25 +2884,28 @@ def bulk_write_to_db(
             session.add(orm_dag)
             orm_dags.append(orm_dag)
 
-        # Get the latest dag run for each existing dag as a single query (avoid n+1 query)
-        most_recent_subq = (
-            session.query(DagRun.dag_id, func.max(DagRun.execution_date).label("max_execution_date"))
-            .filter(
-                DagRun.dag_id.in_(existing_dags),
-                or_(DagRun.run_type == DagRunType.BACKFILL_JOB, DagRun.run_type == DagRunType.SCHEDULED),
+        most_recent_runs = {}
+        num_active_runs = {}
```

Comment:
From past experience, Mypy will probably ask you to provide better type hints for these. Instead of having an empty dict here, it’s generally easier to add an else block.

Reply:
Adjusted code. Thanks for the hint!